common/serial/port_io.h

```diff
@@ -0,0 +1,43 @@
+#ifndef PORT_IO__H
+#define PORT_IO__H
+
+#include <arch.h>
+
+class port_io {
+
+public:
+	port_io(arch_handler *arch) : m_arch(arch) 
+	{
+
+	}
+
+	~port_io(void) 
+	{
+		delete m_arch;
+	}
+
+	void port_write(uint16_t port, uint8_t value)
+	{
+		m_arch->port_write(port, value);
```

Comment:
We will have to talk about how far we want to generalize this stuff. Not sure an arch class is needed. It might be.... not sure. In the end.... the actual port IO will have to come from code written in NASM as I don't want to use inline assembly.